packages/@react-aria/ssr/src/SSRProvider.tsx

```diff
@@ -152,14 +152,15 @@ function useLegacySSRSafeId(defaultId?: string): string {
   }
 
   let counter = useCounter(!!defaultId);
-  return defaultId || `react-aria${ctx.prefix}-${counter}`;
+  let prefix = ctx === defaultContext && process.env.NODE_ENV === 'test' ? 'react-aria' : `react-aria${ctx.prefix}`;
```

Comment:
`process.env` is always available in node, but on the browser-side it may not be defined, depending on the build-tools?